src/pipeline/src/etl/processor.rs

```diff
@@ -264,58 +267,86 @@ fn parse_processor(doc: &yaml_rust::Yaml) -> Result<ProcessorBuilders, String> {
         urlencoding::PROCESSOR_URL_ENCODING => {
             ProcessorBuilders::UrlEncoding(UrlEncodingProcessorBuilder::try_from(value)?)
         }
-        _ => return Err(format!("unsupported {} processor", str_key)),
+        _ => return UnsupportedProcessorSnafu { processor: str_key }.fail(),
     };
 
     Ok(processor)
 }
 
-pub(crate) fn yaml_string(v: &yaml_rust::Yaml, field: &str) -> Result<String, String> {
-    v.as_str()
-        .map(|s| s.to_string())
-        .ok_or(format!("'{field}' must be a string"))
+pub(crate) fn yaml_string(v: &yaml_rust::Yaml, field: &str) -> Result<String> {
+    v.as_str().map(|s| s.to_string()).ok_or(
```

Comment:
not sure why clippy is not jumping in here. But I think we should use `ok_or_else` for lazy-evaluation